manage_containers.py

```diff
@@ -106,18 +110,10 @@ def identify_runtime_cmd(self):
 def build_images(needed_compiler, images):
     """Build container image(s) providing the specified compiler"""
     for c in images:
-        if needed_compiler == 'all':
-            print(f'Add Ubuntu-{c.ubuntu} container image with Clang-{c.clang} and GCC-{c.gcc}')
-            if not c.id:
-                c.build()
-            else:
-                print('[!] WARNING: exists, skipping!')
-        if needed_compiler in ('clang-' + c.clang, 'gcc-' + c.gcc):
-            if c.id:
-                sys.exit(f'[!] ERROR: container image with {needed_compiler} already exists!')
-            print(f'Add Ubuntu-{c.ubuntu} container image with Clang-{c.clang} and GCC-{c.gcc}')
+        if needed_compiler in ('all', 'clang-' + c.clang, 'gcc-' + c.gcc):
             c.build()
-            return
+            if needed_compiler != 'all':
+                return
```

Comment:
Nice addition, a separate error output for the case with multiple and single addition was really unnecessary! I also liked the technique of checking input below, it was not so obvious to me)